### DISCUSSION TITLE
im new in coding can someone gimme suggestion

### DISCUSSION THREAD

**Discussion by andmcpro** [community/community#65423] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

hello im new in coding
can someone help me with gimme some suggestion 
i like coding about android but i dint know what must i choose dart/kotlin can someone explaine me and what u think is better

also i can do basic programming php can someone gimme instructions to advance my php skill

**Comment by seyLu** [comment]
#### Hi @andmcpro 

#### If you're new to coding, I'd suggest checking out [Harvard CS50x](https://cs50.harvard.edu/x/2023/). This will give you a good foundation for Computer Science. 

#### Learn the fundamentals first -- stuff that are laguage-agnostic -- eg. Flow Charts, Data Structures and Algorithms, etc.

#### What programming language you use to learn the fundamentals doesn't matter, but I'd suggest stick to learning a language for at least 6 months before learning your next one.

#### For learning more about PHP and its ecosystem, I highly recommend [Laracasts](https://laracasts.com/). You can take their PHP course for free, and maybe learn the Laravel framework afterwards.

#### A lot of modern tech is using frameworks -- which can be overwhelming to learn if you're not comfortable on using the language that is used to build that said framework.

#### In this case learn PHP first, before learning Laravel. (or JavaScript before React, or C# before Unity, or Java before Spring, you get what I mean)

#### I'm not an android dev myself so I can't really comment on Dart vs Kotlin. By why not learn the former first, then learn the second, and then master which one you like best. 

#### Again, if you have good fundamentals, a solid foundation, whatever tech you learn shouldn't matter. You're free to try things out, get a feel for the tech, see if you enjoy working with it, and continue to learn it if you really like it.

**Comment by Pinedah** [comment]
Hi there!

Choosing Between Dart and Kotlin for Android Development
Which is Better?
If you want to develop for both Android and iOS: Go with Dart and Flutter.
If you are focusing only on Android development: Go with Kotlin.

Advancing Your PHP Skills
Since you already have basic knowledge of PHP, here are some steps to help you advance:

Learn Object-Oriented Programming (OOP):
Understand the concepts of classes, objects, inheritance, polymorphism, and interfaces in PHP.
Practice writing OOP code and try to implement small projects using OOP principles.

Work with Databases:
Deepen your knowledge of SQL and how to interact with databases using PHP.
Learn about PDO (PHP Data Objects) for secure and flexible database interactions.

Explore PHP Frameworks:
Learn and work with popular PHP frameworks like Laravel (personal recommend this one), Symfony, or CodeIgniter.
Frameworks will help you understand MVC (Model-View-Controller) architecture and improve your development speed and code organization.

.. and some more:
Enhance Your Debugging Skills
Learn about PHP Securit
API Development
Explore Advanced Topics

Apply what you've learned by building real-world projects. This could be anything from a simple blog to a complex e-commerce site.
Contributing to open-source projects is also a great way to learn and improve your skills.

By following these steps, you'll be able to advance your PHP skills and become a more proficient developer. Good luck!

**Comment by KhujamovCodes** [comment]
Response #4
